sdk/go/property/map.go

```diff
@@ -105,6 +105,15 @@ func (m Map) Set(key string, value Value) Map {
 	return Map{cp}
 }
 
+// Delete produces a new map identical to the receiver with given keys removed.
+func (m Map) Delete(keys ...string) Map {
+	cp := copyMapNonNil(m.m)
+	for _, k := range keys {
+		delete(cp, k)
+	}
+	return Map{cp}
```

Comment:
Let's make sure that we avoid putting a non-nil empty map in `Map`.
```suggestion
	if len(cp) == 0 {
		return Map{}
	}
	return Map{cp}
```

We should also add a test that `m.Delete("a", "b", "c")` asserts as equal to `Map{}`.